torch/distributions/one_hot_categorical.py

```diff
@@ -76,11 +76,15 @@ def param_shape(self):
     def sample(self, sample_shape=torch.Size()):
         sample_shape = torch.Size(sample_shape)
         probs = self._categorical.probs
-        one_hot = probs.new(self._extended_shape(sample_shape)).zero_()
         indices = self._categorical.sample(sample_shape)
+        if torch._C._get_tracing_state():
+            # [JIT WORKAROUND] lack of support for .scatter_()
+            eye = torch.eye(self.event_shape[-1], dtype=self._param.dtype, device=self._param.device)
+            return eye[indices].clone()
```

Comment:
Why the `clone()`? Can you please add a comment in the code?

Reply:
`eye[indices]` was returning a view, which was triggering a test failure due to in-place `.detach_()`. I'll try to remove the `.clone()` and avoid in-place `.detach_()` in the test.